src/embedded.js

```diff
@@ -1,981 +1,1051 @@
-/**
- * HelloSign JS library for embeddables
- * Copyright (c) 2016 HelloSign
- *
- * XWM - Cross-window messaging inspired by Ben Alman's
- * jQuery postMessage plugin:
- * http://benalman.com/projects/jquery-postmessage-plugin/
- *
- *    Copyright (c) 2009 "Cowboy" Ben Alman
- *    Dual licensed under the MIT and GPL licenses.
- *    http://benalman.com/about/license/
- */
-
-(function(){
-
-    function getUrlVars() {
-        var vars = {};
-        var parts = window.location.href.replace(/[?&]+([^=&]+)=([^&]*)/gi,
-                function(m, key, value) {
-                    vars[key] = value;
-                });
-        return vars;
+import Emitter from 'tiny-emitter';
+import { safeHtml } from 'common-tags';
+
+import debug from './utils/debug';
+import defaults from './defaults';
+import settings from './settings';
+
+class HelloSign extends Emitter {
+
+  /**
+   * HelloSign Embedded class names.
+   *
+   * @enum {string}
+   * @static
+   * @readonly
+   */
+  static classNames = settings.classNames;
+
+  /**
+   * HelloSign Embedded events.
+   *
+   * @enum {string}
+   * @static
+   * @readonly
+   */
+  static events = settings.events;
+
+  /**
+   * HelloSign Embedded supported locales.
+   *
+   * @enum {string}
+   * @static
+   * @readonly
+   */
+  static locales = settings.locales;
+
+  /**
+   * HelloSign Embedded cross-origin window messages.
+   *
+   * @enum {string}
+   * @static
+   * @readonly
+   */
+  static messages = settings.messages;
+
+  /**
+   * HelloSign Embedded version number.
+   *
+   * @enum {string}
+   * @static
+   * @readonly
+   */
+  static version = __PKG_VERSION__;
+
+  /**
+   * The base config object which "open" will extend.
+   *
+   * @type {?Object}
+   * @private
+   */
+  _baseConfig = null;
+
+  /**
+   * A reference to the base HelloSign Embedded container
+   * element.
+   *
+   * @type {?HTMLElement}
+   * @private
+   */
+  _baseEl = null;
+
+  /**
+   * The final config object.
+   *
+   * @type {?Object}
+   * @private
+   */
+  _config = null;
+
+  /**
+   * The embedded flow type.
+   *
+   * @type {?string}
+   * @private
+   */
+  _embeddedType = null;
+
+  /**
+   * The iFrame URL object.
+   *
+   * @type {?URL}
+   * @private
+   */
+  _iFrameURL = null;
```

Comment:
Isn't this a string?

Reply:
Nope! `URL` https://developer.mozilla.org/en-US/docs/Web/API/URL :) I've included a polyfill for it at the webpack level.

Reply:
Oh sweet!